ledgerblue/recoverMutualAuth.py

```diff
@@ -0,0 +1,160 @@
+from ledgerblue.ecWrapper import PublicKey, PrivateKey
+from ledgerblue.hexLoader import HexLoader
+from ledgerblue.recoverSCP import scp_derive_key, extract_from_certificate, decrypt_certificate
+from ecpy.curves import Point, Curve
+from Crypto.Cipher import AES
+from hashlib import sha256
+import binascii
+import struct
+import os
+
+CERT_ROLE_SIGNER = 0x1
+CERT_ROLE_SIGNER_EPHEMERAL = 0x11
+CERT_ROLE_DEVICE = 0x2
+CERT_ROLE_DEVICE_EPHEMERAL = 0x12
+CERT_ROLE_RECOVER_ORCHESTRATOR = 0x05
+CERT_ROLE_RECOVER_ORCHESTRATOR_EPHEMERAL = 0x15
+
+CERT_ROLE_RECOVER_PROVIDER = 0x4
+CERT_ROLE_RECOVER_PROVIDER_EPHEMERAL = 0x14
+CERT_FORMAT_VERSION = 0x01
+
+
+def SCPv3(dongle, issuer_public_key, target_id, ca_private_key=None, signer_private_key=None,
+          user_mode=False):
+    if not user_mode:
+        ca_sk = PrivateKey(bytes(ca_private_key))
+    target = bytearray(struct.pack('>I', target_id))
+    scpv3 = 0x02
+
+    apdu = bytearray([0xe0, 0x04, scpv3, 0x00]) + bytearray([len(target)]) + target
+    dongle.exchange(apdu)
+
+    # Initialize authentication
+    nonce = os.urandom(8)
+    apdu = bytearray([0xe0, 0x50, 0x00, 0x00]) + bytearray([len(nonce)]) + nonce
+
+    auth_info = dongle.exchange(apdu)
+    batch_signer_serial = auth_info[0:4]
```

Comment:
pylint reports this variable is unused, is it normal?

Reply:
Yes, batch_signer_serial is used to select the right public key to verify the device's certificate but it isn't relevant here, so I just removed it.